server/service/src/invoice_line/stock_out_line/update/generate.rs

```diff
@@ -135,9 +137,9 @@ fn generate_line(
         item_variant_id,
         vvm_status_id,
         linked_invoice_id: None,
-        donor_link_id: None,
+        donor_link_id,
```

Comment:
The donor_link_id needs to be passed through otherwise it gets reset to "Null" by repository layer (unrelated to this issue, but discovered while debugging this one with @jmbrunskill )